xray/test/test_backends.py

```diff
@@ -584,6 +584,20 @@ def test_variable_order(self):
         with self.roundtrip(ds) as actual:
             self.assertEqual(list(ds), list(actual))
 
+    def test_unsorted_index_raises(self):
+        random_data = np.random.random(size=(4, 6))
+        dim0 = [0, 1, 2, 3]
+        dim1 = [0, 2, 1, 3, 5, 4]  # We will sort this in a later step
+        da = xray.DataArray(data=random_data, dims=('dim0', 'dim1'),
+                            coords={'dim0': dim0, 'dim1': dim1}, name='randovar')
+        ds = da.to_dataset()
+
+        with self.roundtrip(ds) as ondisk:
```

Comment:
It writes it to disk and then reads it back out, but the new dataset should not be loaded into memory